pyvista/plotting/plotter.py

```diff
@@ -3370,18 +3371,13 @@ def add_mesh(
 
         # Try to plot something if no preference given
         if scalars is None and color is None and texture is None:
-            # Prefer texture first
-            if mesh._textures:
-                texture = True
-            # If no texture, plot any active scalar
+            # Make sure scalars components are not vectors/tuples
+            scalars = mesh.active_scalars_name
+            # Don't allow plotting of string arrays by default
+            if scalars is not None:  # and np.issubdtype(mesh.active_scalars.dtype, np.number):
+                scalar_bar_args.setdefault('title', scalars)
             else:
-                # Make sure scalars components are not vectors/tuples
-                scalars = mesh.active_scalars_name
-                # Don't allow plotting of string arrays by default
-                if scalars is not None:  # and np.issubdtype(mesh.active_scalars.dtype, np.number):
-                    scalar_bar_args.setdefault('title', scalars)
-                else:
-                    scalars = None
+                scalars = None
```

Comment:
The inherited `else` seems redundant here:
```py
if scalars is not None:
    ...
else:
    scalars = None
 ```

And I'm a bit wary of the original logic here:
```py
                if scalars is not None:  # and np.issubdtype(mesh.active_scalars.dtype, np.number):
                    scalar_bar_args.setdefault('title', scalars)
                else:
                    scalars = None
                scalars = None
```
This looks to me that even in the `scalars is not None` case we replace the value with `salars=None`. This no longer happens now. Any surprises downstream?

Reply:
The GitHub diff isn't doing us any favor in reviewing this... I agree on the first point about redundancy, but the change here is simply getting rid of the logic for `mesh._textures` and de-indenting everything that used to be in the `else` block.  I think it is fine to keep the ancillary changes to a minimum, i.e. don't address the redudancy, or clean up here.

I don't understand the second point as the second `scalars=None` line doesn't exist in the original logic, at least not within this section of the code that I can see.

Reply:
Ah, you're right, I got confused by the convoluted diff. I agree that just unindenting is the way to go. I thought exactly that this wasn't what was happening.